cppForSwig/TxClasses.cpp

```diff
@@ -371,22 +371,34 @@ bool Tx::isCoinbase(void) const
 /////////////////////////////////////////////////////////////////////////////
 void Tx::unserialize(uint8_t const * ptr, size_t size)
 {
-   uint32_t nBytes = BtcUtils::TxCalcLength(ptr, size, &offsetsTxIn_, &offsetsTxOut_);
+	uint32_t nBytes = BtcUtils::TxCalcLength(ptr, size, &offsetsTxIn_, &offsetsTxOut_, &offsetsWitness_);
 
-   if (nBytes > size)
-      throw BlockDeserializingException();
-   dataCopy_.copyFrom(ptr, nBytes);
-   BtcUtils::getHash256(ptr, nBytes, thisHash_);
-   if (8 > size)
-      throw BlockDeserializingException();
+	if (nBytes > size)
+		throw BlockDeserializingException();
+	dataCopy_.copyFrom(ptr, nBytes);
+	if (8 > size)
+		throw BlockDeserializingException();
 
-   uint32_t numTxOut = offsetsTxOut_.size() - 1;
-   version_ = READ_UINT32_LE(ptr);
-   if (4 > size - offsetsTxOut_[numTxOut])
-      throw BlockDeserializingException();
-   lockTime_ = READ_UINT32_LE(ptr + offsetsTxOut_[numTxOut]);
+	uint32_t numTxOut = offsetsTxOut_.size() - 1;
+	version_ = READ_UINT32_LE(ptr);
+	if (4 > size - offsetsTxOut_[numTxOut])
```

Comment:
this check doesn't work anymore, you need to check against the end of the witness data. this means you need to set offsetWitness_ in all cases (with or without witness data), and check 
(4 > size - offsetsWitness_[numWitness])
where numWitness is
offsetWitness_.size() - 1;

This implies offsetWitness is set to offsetsTxOut[numTxOut] where there is no witness data.